src/core/src/package_managers/YumPackageManager.py

```diff
@@ -347,26 +386,249 @@ def get_package_size(self, output):
+            self.env_layer.file_system.write_with_retry(self.os_patch_configuration_settings_file_path, '{0}'.format(updated_patch_configuration_sub_setting.lstrip()), mode='w+')
+        except Exception as error:
+            error_msg = "Error occurred while updating system configuration settings for auto OS updates. [Patch Configuration={0}] [Error={1}]".format(str(patch_configuration_sub_setting), repr(error))
+            self.composite_logger.log_error(error_msg)
+            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.DEFAULT_ERROR)
+            raise
+
+    def disable_auto_update_on_reboot(self, command):
+        self.composite_logger.log_debug("Disabling auto update on reboot using command: " + str(command))
+        code, out = self.env_layer.run_command_output(command, False, False)
+        self.composite_logger.log_debug(" - Code: " + str(code) + ", Output: \n|\t" + "\n|\t".join(out.splitlines()))
+
+        if code != 0:
+            self.composite_logger.log('[ERROR] Command invoked: ' + command)
+            self.telemetry_writer.write_execution_error(command, code, out)
+            error_msg = 'Unexpected return code (' + str(code) + ') on command: ' + command
+            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.OPERATION_FAILED)
+            raise Exception(error_msg, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
+
+        self.composite_logger.log_debug("Auto update on reboot disabled")
+
+    def is_auto_update_service_installed(self, install_check_cmd):
+        """ Checks if the auto update service is enable_on_reboot on the VM """
+        self.composite_logger.log_debug("Checking if auto update service is installed...")
+        code, out = self.env_layer.run_command_output(install_check_cmd, False, False)
+        self.composite_logger.log_debug(" - Code: " + str(code) + ", Output: \n|\t" + "\n|\t".join(out.splitlines()))
+        if len(out.strip()) > 0 and code == 0:
+            self.composite_logger.log_debug("Auto OS update is installed on the machine")
+            return True
+        else:
+            self.composite_logger.log_debug("Auto OS update is NOT installed on the machine")
+            return False
+
+    def install_auto_update_service(self, command):
+        self.composite_logger.log_debug("Installing auto updates using command: " + str(command))
+        code, out = self.env_layer.run_command_output(command, False, False)
+        if code != 0:
+            self.composite_logger.log('[ERROR] Package manager was invoked using: ' + command)
+            self.composite_logger.log_warning(" - Return code from package manager: " + str(code))
+            self.composite_logger.log_warning(" - Output from package manager: \n|\t" + "\n|\t".join(out.splitlines()))
```

Comment:
Reduce to a single line to reduce log load. This will work because return code is just a number.

Reply:
This part of the code no longer exists.